### ISSUE TITLE
Split reducers into different files, or change state logic

### ISSUE THREAD

**Issue by leftmove** [leftmove/wallstreetlocal#7]
Currently, wallstreetlocal uses Redux to manage all the complex, global state for different filers. This leads to a couple problems.

- All reducers are housed in a single, big file, and it is hard to manage them
- Things tend to get unorganized when creating new reducers, they should be more compartmentalized
- Performance can be slow when loading many reducers
- Lots of boilerplate, including complex, and perhaps unnecessary Redux logic

Redux settings probably need to be tweaked to implement more best practices. `filerSlice.js` is a big file, and loading it dynamically, if possible, could improve performance. 

Or maybe the best solution is to abandon Redux altogethor, and adopt a new (or no) state library.

**Comment by leftmove** [comment]
Absolutely! Depending on your experience, this issue may be a bit complex and will take a while to understand.

Let me know which technologies you are most experienced in, and I can point you in the right direciton.

@shubham9069

**Comment by shubham9069** [comment]
@leftmove I worked On MERN as well as had experience in Angular. you can checkout my portfolio which gives me brief explanation of me and my work https://shubhamkaushik-portfolio.vercel.app/

**Comment by leftmove** [comment]
It seems you are a seasoned React developer; this gives you a couple of options depending on what you want to dip your toes in.

- This issue requires fairly in depth knowledge of [Redux.js](https://redux.js.org/) - a state management tool. I would not reccomend it unless you really want to learn all about Redux (especially since many other state management libraries are trending up nowadays).
- [Create charts](https://github.com/leftmove/wallstreetlocal/issues/3). This is the issue I would reccomend the most to you. Basically, you have to take in data and use your knowledge of JavaScript and React to create charts for that data. It would be good practice for any React developer.
- [https://github.com/leftmove/wallstreetlocal/issues/6](Migration to Next.js app router). This is the easiest but most tedious issue; I would also reccomend it. Basically, you just have to update all the dependencies and get the entire app running on Next.js's newest version. It will require lots of refactoring, but there are many guides.

Whatever you choose, please let me know and do not hesitate to ask for help. I'll assign you as soon as possible and give you some tips to start off.

Thank you!

**Comment by shubham9069** [comment]
@leftmove I understand your concern but redux is also built on top of content API and I have prophecy in state Management using Context API. and second yeah I Would Love to contribute other 2 tasks

**Comment by mustafa-tola** [comment]
I would love to work on this task as i have previously worked on redux

**Comment by aadarsh10c** [comment]
Hey @leftmove , looks like person assigned this issue is not working, i would like to contribute to this  issue. Please assign me this issue

**Comment by leftmove** [comment]
@mustafa-tola @aadarsh10c

Sorry for the late reply, have at it.

**Comment by mustafa-tola** [comment]
@leftmove  i have done some work regarding this issue how can i test it sorry for the delay

**Comment by leftmove** [comment]
@mustafa-tola Easiest way to test is to run the front-end with `npm run dev` and make sure all the components are working.

**Comment by mustafa-tola** [comment]
@leftmove i am making a new branch upon this issue. can you test it throughly if it is working correctly or not. i have solved all pf the errors and divided your current reducer into 3 different ones and merged them eventually
